packages/fxa-auth-server/test/local/payments/paypal-processor.js

```diff
@@ -605,7 +605,11 @@ describe('PaypalProcessor', () => {
           yield invoice;
         },
       });
-      await processor.processInvoices();
+      // eslint-disable-next-line
+      for await (const _ of processor.processInvoices()) {
+        // No value yield'd; yielding control for potential distributed lock
+        // extension in actual use case
```

Comment:
This is because Redis lock has a ticker that automatically renews it, right? But processing each invoice involves multiple async steps which will yield to the ticker, so is this needed?

Reply:
No, the lock is auto release after the during for which it was set.  (Default is five minutes.)  That's why the patch periodically extends it.

Reply:
Redlock does have an auto-renew option.  IIRC that'd be cleaner if we didn't have the no lock option for the script itself.  Let me see what I can wrangle up with that approach...